### ISSUE TITLE
index_buffer_size cannot be set

### ISSUE THREAD

**Issue by prog8** [elastic/elasticsearch#18000]
I added this setting

`indices.memory.index_buffer_size: "50%` to elasticsearch.yaml file but I get exception on ES start.
ES 5.0 alpha taken from ES site.

```
[2016-04-27 01:42:37,727][ERROR][bootstrap                ] Exception
java.lang.IllegalArgumentException: unknown setting [indices.memory.index_buffer_size]
        at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:267)
        at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:238)
        at org.elasticsearch.common.settings.SettingsModule.configure(SettingsModule.java:120)
        at org.elasticsearch.common.inject.AbstractModule.configure(AbstractModule.java:60)
        at org.elasticsearch.common.inject.spi.Elements$RecordingBinder.install(Elements.java:233)
        at org.elasticsearch.common.inject.spi.Elements.getElements(Elements.java:103)
        at org.elasticsearch.common.inject.InjectorShell$Builder.build(InjectorShell.java:148)
        at org.elasticsearch.common.inject.InjectorBuilder.build(InjectorBuilder.java:99)
        at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:96)
        at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:70)
        at org.elasticsearch.common.inject.ModulesBuilder.createInjector(ModulesBuilder.java:46)
        at org.elasticsearch.node.Node.<init>(Node.java:229)
        at org.elasticsearch.node.Node.<init>(Node.java:161)
        at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:189)
        at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:263)
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:111)
        at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:106)
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:88)
        at org.elasticsearch.cli.Command.main(Command.java:53)
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:74)
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:67)
```

**Comment by prog8** [comment]
Maybe related to this commit: https://github.com/elastic/elasticsearch/commit/e91a141233d0231fa3f43320968686b602d414e3

**Comment by jasontedor** [comment]
Thanks for reporting! This is fixed in #17778. I'll add you to the [pioneer program](https://www.elastic.co/blog/elastic-pioneer-program).
